laurat007/assignment1/Q2.java

```diff
@@ -0,0 +1,95 @@
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author laura
+ * The following program will get the K-th to last element in a linked list.
+ * It reads the input from a file containing on the first line a number N meaning the length of the list,
+ * and on the second line N numbers meaning the elements of the list.
+ */
+public class Q2 {
+	
+	public static final String INPUT_FILE = "in";
+	
+	static class MyList {
+		Object obj;
```

Comment:
Have you heard about java generics? Can you try and use them here? This would allow you to have a MyList of ints or MyList of Strings and use them as typed objects.

Reply:
Not sure if I added generics correctly.